pallets/order-book/src/lib.rs

```diff
@@ -414,18 +428,100 @@ pub mod pallet {
+
+            // min <= max
+            // It is possible to set min == max if it necessary, e.g. some NFTs
+            ensure!(min_lot_size <= max_lot_size, Error::<T>::InvalidMaxLotSize);
+
+            if T::AssetInfoProvider::is_non_divisible(&order_book_id.base) {
+                // NFT has special bounds as non-divisible asset
+                ensure!(step_lot_size >= balance!(1), Error::<T>::InvalidStepLotSize);
+                ensure!(
+                    step_lot_size % balance!(1) == 0,
+                    Error::<T>::InvalidStepLotSize
+                );
+            }
+
+            // min & max couldn't be less then `step_lot_size`
+            ensure!(min_lot_size >= step_lot_size, Error::<T>::InvalidMinLotSize);
+            ensure!(max_lot_size >= step_lot_size, Error::<T>::InvalidMaxLotSize);
+
+            // min & max must be a multiple of `step_lot_size`
+            ensure!(
+                min_lot_size % step_lot_size == 0,
+                Error::<T>::InvalidMinLotSize
+            );
+            ensure!(
+                max_lot_size % step_lot_size == 0,
+                Error::<T>::InvalidMaxLotSize
+            );
+
+            // Even if `tick_size` & `step_lot_size` meet precision conditions the min possible deal amount could not match.
+            // The min possible deal amount = `tick_size` * `step_lot_size`.
+            // We need to be sure that the value doesn't overflow Balance if `tick_size` & `step_lot_size` are too big
+            // and doesn't go out of precision if `tick_size` & `step_lot_size` are too small.
+
+            // Returns error if value overflows.
+            let min_possible_deal_amount = (FixedWrapper::from(tick_size)
+                * FixedWrapper::from(step_lot_size))
+            .try_into_balance()
+            .map_err(|_| Error::<T>::TickSizeAndStepLotSizeAreTooBig)?;
+
+            // 1 is a min non-zera possible value. balance!(0.000000000000000001) == 1
+            // If `tick_size` * `step_lot_size` result goes out of 18 digits precision, the min possible deal amount == 0,
+            // because FixedWrapper::try_into_balance() returns 0 for such cases.
+            ensure!(
+                min_possible_deal_amount >= 1,
+                Error::<T>::TickSizeAndStepLotSizeAreTooSmall
+            );
+
+            // `max_lot_size` couldn't be more then total supply of `base` asset
+            let total_supply = T::AssetInfoProvider::total_issuance(&order_book_id.base)?;
+            ensure!(
+                max_lot_size <= total_supply,
+                Error::<T>::MaxLotSizeIsMoreThanTotalSupply
+            );
+
+            order_book.tick_size = tick_size;
+            order_book.step_lot_size = step_lot_size;
+            order_book.min_lot_size = min_lot_size;
+            order_book.max_lot_size = max_lot_size;
+
+            // Note:
+            // Already existed limit orders are not changed even if they don't meet the requirements of new attributes.
```

Comment:
The only parameter change that might affect working with existing orders is `step_lot_size` which is used in `align_amount`.

As I see, it shouldn't break anything, but just want to confirm: it's completely ok, isn't it?

Reply:
There is 1 scenario that we should investigate

Reply:
Will be handled in #498